### PR TITLE
add label selector in Resource Modifiers 

### PR DESCRIPTION
**Pull request by 27149chen**
# Please add a summary of your change

This pr made some improvements in Resource Modifiers:
1. add label selector
2. change the field name from groupKind to groupResource

# Does your change fix a particular issue?

N/A

# Please indicate you've done the following:

- [x] [Accepted the DCO](https://velero.io/docs/v1.5/code-standards/#dco-sign-off). Commits without the DCO will delay acceptance.
- [x] [Created a changelog file](https://velero.io/docs/v1.5/code-standards/#adding-a-changelog) or added `/kind changelog-not-required` as a comment on this pull request.
- [ ] Updated the corresponding documentation in `site/content/docs/main`.

### CODE REVIEW

File: changelogs/unreleased/6704-27149chen [hunk 1]
```diff
@@ -0,0 +1,3 @@
+This pr made some improvements in Resource Modifiers:
+1. add label selector
+2. change the field name from groupKind to groupResource
```
Comment (anshulahuja98) [inline changelogs/unreleased/6704-27149chen:3]: can you please explain your rationale for the change to groupResource?
I preferred groupKind for clarity in K8s terms.
is there any precedence for groupResource?

File: internal/resourcemodifiers/resource_modifiers_test.go [hunk 2]
```diff
@@ -18,7 +18,7 @@ func TestGetResourceModifiersFromConfig(t *testing.T) {
 			Namespace: "test-namespace",
 		},
 		Data: map[string]string{
-			"sub.yml": "version: v1\nresourceModifierRules:\n- conditions:\n    groupKind: persistentvolumeclaims\n    resourceNameRegex: \".*\"\n    namespaces:\n    - bar\n    - foo\n  patches:\n  - operation: replace\n    path: \"/spec/storageClassName\"\n    value: \"premium\"\n  - operation: remove\n    path: \"/metadata/labels/test\"\n\n\n",
+			"sub.yml": "version: v1\nresourceModifierRules:\n- conditions:\n    groupResource: persistentvolumeclaims\n    resourceNameRegex: \".*\"\n    namespaces:\n    - bar\n    - foo\n  patches:\n  - operation: replace\n    path: \"/spec/storageClassName\"\n    value: \"premium\"\n  - operation: remove\n    path: \"/metadata/labels/test\"\n\n\n",
```
Comment (anshulahuja98) [inline internal/resourcemodifiers/resource_modifiers_test.go:21]: can you add a deserialization test for labels?

File: site/content/docs/v1.12/restore-resource-modifiers.md [hunk 3]
```diff
@@ -80,7 +80,7 @@ resourceModifierRules:
 version: v1
 resourceModifierRules:
 - conditions:
-    groupKind: deployments.apps
+    groupResource: deployments.apps
```
Comment (anshulahuja98) [inline site/content/docs/v1.12/restore-resource-modifiers.md:83]: can you add docs for label selector example.

**Review by anshulahuja98 — CHANGES_REQUESTED** [review]

File: internal/resourcemodifiers/resource_modifiers_test.go (hunk 2 above)
Comment (27149chen) [reply internal/resourcemodifiers/resource_modifiers_test.go:21]: added

File: site/content/docs/v1.12/restore-resource-modifiers.md (hunk 3 above)
Comment (27149chen) [reply site/content/docs/v1.12/restore-resource-modifiers.md:86]: added

File: changelogs/unreleased/6704-27149chen (hunk 1 above)
Comment (anshulahuja98) [reply changelogs/unreleased/6704-27149chen:3]: got it thanks for clarifying.

**Review by anshulahuja98 — DISMISSED** [review]

**Comment by anshulahuja98** [comment]
CC @reasonerjt, we need to put this in 1.12 release branch, do you folks generally handle the cherry pick of such changes to release branch, or can I help in any way here?

File: site/content/docs/v1.12/restore-resource-modifiers.md [hunk 4]
```diff
@@ -42,9 +45,9 @@ resourceModifierRules:
     path: "/metadata/labels/test"
  ```
 
-- The above configmap will apply the JSON Patch to all the PVCs in the namespaces bar and foo with name starting with mysql. The JSON Patch will replace the storageClassName with "premium" and remove the label "test" from the PVCs.
+- The above configmap will apply the JSON Patch to all the PVCs in the namespaces bar and foo with name starting with mysql and match label `foo: bar`. The JSON Patch will replace the storageClassName with "premium" and remove the label "test" from the PVCs.
```
Comment (ywk253100) [inline site/content/docs/v1.12/restore-resource-modifiers.md:48]: Please also make the same changes to the doc in the main branch

**Review by ywk253100 — APPROVED** [review]

**Review by anshulahuja98 — APPROVED** [review]

**Pull request merged**